src/Psalm/Baseline.php

```diff
@@ -0,0 +1,159 @@
+<?php
+namespace Psalm;
+
+class Baseline
+{
+    /**
+     * @param string $baselineFile
+     * @param array<array{file_name: string, type: string}> $issues
+     *
+     * @return void
+     */
+    public static function create(string $baselineFile, array $issues)
+    {
+        $groupedIssues = self::countIssueTypesByFile($issues);
+
+        self::writeToFile($baselineFile, $groupedIssues);
+    }
+
+    /**
+     * @param string $baselineFile
+     * @return array<string,array<string,int>>
+     * @throws Exception\ConfigException
+     */
+    public static function read(string $baselineFile): array
+    {
+        if (!file_exists($baselineFile) || !is_readable($baselineFile)) {
+            throw new Exception\ConfigException("{$baselineFile} does not exist or is not readable\n");
+        }
+
+        $baselineDoc = new \DOMDocument();
+        $baselineDoc->load($baselineFile, LIBXML_NOBLANKS);
+
+        /** @var \DOMNodeList $filesElement */
+        $filesElement = $baselineDoc->getElementsByTagName('files');
+
+        if ($filesElement->length === 0) {
+            throw new Exception\ConfigException('Baseline file does not contain <files>');
+        }
+
+        $files = [];
+
+        /** @var \DOMElement $filesElement */
+        $filesElement = $filesElement[0];
+
+        /** @var \DOMElement $file */
+        foreach ($filesElement->getElementsByTagName('file') as $file) {
+            $fileName = $file->getAttribute('src');
+
+            $files[$fileName] = [];
+
+            /** @var \DOMElement $issue */
+            foreach ($file->childNodes as $issue) {
+                $issueType = $issue->tagName;
+
+                $files[$fileName][$issueType] = (int)$issue->getAttribute('occurrences');
+            }
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param string $baselineFile
+     * @param array<array{file_name: string, type: string}> $issues
+     * @return array<string,array<string,int>>
+     * @throws Exception\ConfigException
+     */
+    public static function update(string $baselineFile, array $issues)
+    {
+        $existingIssues = self::read($baselineFile);
+        $newIssues = self::countIssueTypesByFile($issues);
+
+        foreach ($existingIssues as $file => &$existingIssuesCount) {
+            if (!isset($newIssues[$file])) {
+                unset($existingIssues[$file]);
```

Comment:
Would you be able to add a test case for this? It's the only block not covered

Reply:
Yes, good one 👍  had that covered before but missed another case, thought I fixed that, but didn't verify because the coverage run took too long.